### ISSUE TITLE
VersionTask adds trailing new line to the file

### ISSUE THREAD

**Issue by chrisjohnson00** [phingofficial/phing#711]
When using the VersionTask to increment a version file, I'm finding that it's adding a trailing new line to the version file.  Elsewhere in my build.xml I read from this file (in cases where I need the version number, without incrementing it - deploying the latest build), and at this point it's outputting the version number with an extra line break.  If using the version number, in a file name for example, it breaks the file name by adding the trailing new line.

Here's the version task I'm using

    <version releasetype="${release_type}" file="version" property="version.number"/>

When I start with file with `0.0.0` as the contents of this file (single line file)
After running the above task, I now get `0.1.0` but with a new line character at the end.

My expectation is that a new line character is not added.

This causes the following to not work

        <if>
            <isset property="version.number"/>
            <then/>
            <else>
                <echo msg="Loading version number from version file"/>
                <loadfile property="version.number" file="version"/>
            </else>
        </if>
        <echo msg="Deploying version ${version.number} to ${deploy_host}"/>
        <scp username="${deploy_username}" password="${deploy_password}" host="${deploy_host}" todir="/tmp/">
            <fileset dir="${project.basedir}/${builddir}">
                <include name="${package}_${version.number}.tar.gz"/>
            </fileset>
        </scp>
        <echo msg="Untaring the build into the application directory"/>
        <ssh username="${deploy_username}" password="${deploy_password}" host="${deploy_host}" command="tar xzf /tmp/${package}_${version.number}.tar.gz -C /${deploy_application_path}/" failonerror="true"/>
        
The `scp` task results in 0 files transferred.  And the final task to untar the file on the remote, fails with the following error

    BUILD FAILED
    /vagrant/build.xml:114:205: SSH Task failed: tar (child): /tmp/cron_manager_0.2.0: Cannot open: 
    No such file or directory

In the error output you can see that the filename is breaking into a new line, causing the SSH command to terminate prematurely.

Suggested fix - do not add a new line character to the end of the version file from the VersionTask

**Comment by chrisjohnson00** [comment]
see PR #713

**Comment by chrisjohnson00** [comment]
if anyone lands on this issue and my PR isn't in a live version, or is declined by the project owners... here's the workaround I had to put in for this issue.

                <loadfile property="version.number" file="version"/>
                <php function="trim" returnProperty="version.number">
                    <param value="${version.number}"/>
                </php>

**Comment by siad007** [comment]
There is a (not documented) attribute `propfile` in the `VersionTask` to save property file of your version
```xml
<version releasetype="${release_type}" file="version" property="version.number" propfile="true"/>
<echo msg="Load version from property-file"/>
<property file="version"/>
```
But this should be added to the documentation as well.

**Comment by chrisjohnson00** [comment]
@siad007 i have use cases where I'm creating a tar.gz of my project without incrementing the version in the version file, just reading the value from the file.  Thus, I'm using `loadfile`to read from the version file without altering the version number.  This is where the issue resides, the value in my version file has a trailing line break, which gets carried over into the property value when using loadfile.

**Comment by siad007** [comment]
Why not just reading from a property file instead?

**Comment by chrisjohnson00** [comment]
@siad007 I tried adding `propfile="true"` into my version task, but that resulted in the following error

```
BUILD FAILED
/vagrant/build.xml:87:12: /vagrant/build.xml:91:116: phing.tasks.ext.VersionTask doesn't support the 'propfile' attribute.
```

I'm running `Phing 2.16.0`.

**Comment by siad007** [comment]
@chrisjohnson00 uh oh - my fault - i didn't recognice your version of phing. Sorry for your inconvinence there and you are absolutely right to have such workaround for your phing version or lower. I just have the actual master code in my scope, where it would be possible this kind of the behavior you mentioned could be made in a more better 'phing' way. Imagine you not only using this version file generated by the normal version task way to have a version manifested and persistented in a file for your build/delivery workflow, but you want to append additional information generated by other custom targets/tasks to this file (yeah i know it's very constructed) so you don't have to boger of an additional line break before concatinate both. I know what you mean but i think using phing with the actual master/3.0 version would be to save your information first in a parseable file format (i.e. property file) which could be then used getting the information you want without having the 'oh there is a linebreak which i did not want' think in mind.
What i mean or wanted to explain is that a generated version file could be in a chain of other tasks that interact with this file, where it would be easier (or wanted) that a line break is given by default.
In your usecase, it would be better to save your information in a general better parseble file format to get the needed info without doing a transformation of any kind.

**Comment by siad007** [comment]
This is included in the upcoming phing 3.0 release. So i will close this issue.
